### PR TITLE
@alloy => show the history before a query on quicksilver - fixes #98

### LINKED ISSUE #98: Quicksilver should show last clicked options by default
screen shot 2015-02-03 at 4 27 05 pm

### PR DESCRIPTION
**Pull request by orta**
With the last feature I made, most of my time was spent getting back to the model that was off. This would have made the iteration time significantly shorter.

selfie-1

screen shot 2015-02-03 at 6 12 51 pm

### CODE REVIEW

File: Artsy/Classes/View Controllers/ARQuicksilverViewController.m [hunk 1]
```diff
@@ -16,21 +18,37 @@ @implementation ARQuicksilverViewController
...
 
 #pragma mark - UIViewController
 
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+
+    NSString *path = [ARFileUtils appDocumentsPathWithFolder:@"dev" filename:@"quicksilver_history"];
+    NSData *data = [NSData dataWithContentsOfFile:path];
+    _resultsHistory = [NSKeyedUnarchiver unarchiveObjectWithData:data];
```
Comment (alloy) [inline Artsy/Classes/View Controllers/ARQuicksilverViewController.m]: As I understand it, we’re moving all access to ivars to getters/setters, so why are you directly accessing `_resultsHistory` here?

File: Artsy/Classes/View Controllers/ARQuicksilverViewController.m [hunk 2]
```diff
@@ -16,21 +18,37 @@ @implementation ARQuicksilverViewController
...
+
+    if (!self.resultsHistory) {
+        _resultsHistory = @[];
```
Comment (alloy) [inline Artsy/Classes/View Controllers/ARQuicksilverViewController.m]: Ditto

File: Artsy/Classes/View Controllers/ARQuicksilverViewController.m [hunk 3]
```diff
@@ -152,9 +184,20 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
+- (NSArray *)contentArray
+{
+   return (self.searchBar.text.length == 0) ? self.resultsHistory : self.searchResults;
+}
+
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    SearchResult *result = self.searchResults[indexPath.row];
+    SearchResult *result = self.contentArray[indexPath.row];
+    _resultsHistory = [self.resultsHistory arrayByAddingObject:result];
```
Comment (alloy) [inline Artsy/Classes/View Controllers/ARQuicksilverViewController.m]: Ditto

**Comment by alloy** [comment]
I guess this doesn’t get a CHANGELOG entry because it’s a developer-only thing?

**Comment by alloy** [comment]
Besides those direct accesses to the ivar it looks good to me :+1:

File: Artsy/Classes/View Controllers/ARQuicksilverViewController.m (hunk 1 above)
Comment (orta) [inline Artsy/Classes/View Controllers/ARQuicksilverViewController.m]: it's a `readonly` property, so setting is done via the iVar :+1:

Comment (alloy) [inline Artsy/Classes/View Controllers/ARQuicksilverViewController.m]: There’s a private copying setter defined [here](https://github.com/artsy/eigen/pull/101/files#diff-6221d5ce48ef704ecd76ec92f867e5efR11)?

Comment (orta) [inline Artsy/Classes/View Controllers/ARQuicksilverViewController.m]: oh yeah, that's a bug :D

I should move them all to `readonly`

Comment (ashfurrow) [inline Artsy/Classes/View Controllers/ARQuicksilverViewController.m]: `readonly` in the .m file? Any reason not to keep it read/write privately?

**Comment by orta** [comment]
Alrighty, I've dealt with the property naming stuff :+1: - making things that are changed often `readwrite`, and things that should be immutable `readonly`

**Comment by orta** [comment]
and yeah, no changelog, not really useful for anyone else.

**Pull request merged**